plugin-manager/src/main/kotlin/com/itsaky/androidide/plugins/manager/services/IdeProjectServiceImpl.kt

```
@@ -2,125 +2,200 @@
+
+		fun getProjectByPath(path: File): IProject?
+	}
+
+	override fun getCurrentProject(): IProject? {
+		if (!hasRequiredPermissions()) {
+			throw SecurityException("Plugin $pluginId does not have required permissions: ${getRequiredPermissionsString()}")
+		}
+
+		return try {
+			projectProvider.getCurrentProject()
+		} catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)